### PR TITLE
refactor: improve initialization of prometheus storage spec

### PR DESCRIPTION
**Pull request by sthaha**
Signed-off-by: Sunil Thaha <sthaha@redhat.com>

### CODE REVIEW

File: pkg/controllers/monitoring-stack/components_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		{pvc: nil, expected: nil},
+		{pvc: &corev1.PersistentVolumeClaimSpec{}, expected: nil},
+		{
+			pvc: validPVCSpec,
+			expected: &monv1.StorageSpec{
+				VolumeClaimTemplate: v1.EmbeddedPersistentVolumeClaim{
+					Spec: *validPVCSpec,
+				},
+			},
+		},
```
Comment (arajkumar) [inline pkg/controllers/monitoring-stack/components_test.go:38]: nit:
Can you add another test for a valid PersistentVolumeClaimSpec?

**Review by arajkumar — APPROVED** [review]
lgtm

File: pkg/controllers/monitoring-stack/components_test.go (hunk 1 above)
Comment (sthaha) [reply pkg/controllers/monitoring-stack/components_test.go:38]: closing this :)

**Review by jan--f — APPROVED** [review]
:+1:

**Pull request merged**